geomstats/connection.py

```diff
@@ -64,13 +74,35 @@ def __init__(self, metric):
         self.metric = metric
         self.dimension = metric.dimension
 
+    def metric_matrix(self, base_point):
+        metric_matrix = self.metric.inner_product_matrix(base_point)
+        return metric_matrix
+
+    def cometric_matrix(self, base_point):
+        metric_matrix = self.metric_matrix(base_point)
+        cometric_matrix = gs.linalg.inv(metric_matrix)
+        return cometric_matrix
+
+    def metric_derivative(self, base_point):
+        metric_derivative = autograd.jacobian(self.metric_matrix)
+        return metric_derivative(base_point)
+
     def christoffel_symbols(self, base_point):
         """
         Christoffel symbols associated with the connection.
         """
-        # TODO(nina): implement with automatic differentiation.
-        raise NotImplementedError(
-                'The Christoffel symbols are not implemented.')
+        term_1 = gs.einsum('nim,nmkl->nikl',
+                           self.cometric_matrix(base_point),
```

Comment:
no reference found when typing cometric matrix on the interwebs. Could you potentially link a paper or something to enlighten regular citizens with what is a cometric matrix?